src/selectors/entities/groups.ts

```diff
@@ -90,3 +90,26 @@ export const getGroupsAssociatedToChannel = reselect.createSelector(
         return Object.entries(allGroups).filter(([groupID]) => channelGroupIDSet.has(groupID)).map((entry) => entry[1]);
     }
 );
+
+export const getGroupsAssociatedToTeamForReference = reselect.createSelector(
+    getAllGroups,
+    (state: GlobalState, teamID: string) => getTeamGroupIDSet(state, teamID),
+    (allGroups, teamGroupIDSet) => {
+        return Object.entries(allGroups).filter(([groupID]) => teamGroupIDSet.has(groupID)).filter((entry) => entry[1].allow_reference).map((entry) => entry[1]);
+    }
+);
+
+export const getGroupsAssociatedToChannelForReference = reselect.createSelector(
+    getAllGroups,
+    (state: GlobalState, channelID: string) => getChannelGroupIDSet(state, channelID),
+    (allGroups, channelGroupIDSet) => {
+        return Object.entries(allGroups).filter(([groupID]) => channelGroupIDSet.has(groupID)).filter((entry) => entry[1].allow_reference).map((entry) => entry[1]);
+    }
+);
+
+export const getGroupsForReference = reselect.createSelector(
+    getAllGroups,
+    (allGroups) => {
+        return Object.entries(allGroups).filter((entry) => entry[1].allow_reference).map((entry) => entry[1]);
+    }
+);
```

Comment:
I believe there are 4 different sets of groups that will be retrieved:

||Channel Group Synced|Team Group Synced|Set of Groups|
|---|:---:|:---:|---|
|1|F|F|all allowReferences groups|
|2|F|T|all allowReferences groups associated to team (TG)|
|3|T|F|all allowReferences groups associated to channel (CG)|
|4|T|T|union of TG and CG|

Cases 1, 2, and 3 are covered with these methods. Is the plan to do case 4 (the union) on the front-end or did you want to add a selector for that? I'm 0/5.

Reply:
Thanks, added the union (#4) to the front-end (textbox/index.js)